### PR TITLE
test: add ui tests for applications

### PR DESCRIPTION
**Pull request by xiaoyijun**
## Summary

test: add ui tests for applications

## Testing

IT

## Checklist

- [ ] `.changeset`
- [ ] unit tests
- [x] integration tests
- [ ] docs

OR

- [x] This PR is not applicable for the checklist

### CODE REVIEW

File: packages/integration-tests/src/tests/ui/applications/application-test-cases.ts [hunk 1]
```diff
@@ -0,0 +1,252 @@
...
+  };
+  redirectUri?: string;
+  postSignOutRedirectUri?: string;
+};
+
+const nextJs: ApplicationCase = {
+  framework: 'Next.js',
+  name: 'Next.js App',
+  description: 'This is a Next.js app',
+  guideFilename: 'web-next',
```
Comment (simeng-li) [inline packages/integration-tests/src/tests/ui/applications/application-test-cases.ts:18]: should we read these data from a shared resource. What if the content has been updated?

File: packages/integration-tests/src/tests/ui/applications/helpers.ts [hunk 2]
```diff
@@ -0,0 +1,136 @@
...
+) => {
+  // Expect the creation form to be open
+  await expect(page).toMatchElement(
+    '.ReactModalPortal div[class$=header] div[class$=titleEllipsis]',
+    {
+      text: 'Create Application',
+    }
+  );
+
+  await expect(page).toFillForm('form', {
```
Comment (simeng-li) [inline packages/integration-tests/src/tests/ui/applications/helpers.ts:42]: Should we narrow down the form selector within the CreateApplication Modal?

File: packages/integration-tests/src/tests/ui/applications/helpers.ts [hunk 3]
```diff
@@ -0,0 +1,136 @@
...
+  page: Page,
+  { guideFilename, sample, redirectUri, postSignOutRedirectUri }: ApplicationCase,
+  skipFillForm = false
+) => {
+  await expect(page).toMatchElement(
+    '.ReactModalPortal div[class$=header] div[class$=titleEllipsis]',
+    {
+      text: 'Start with SDK and guides',
+    }
+  );
```
Comment (simeng-li) [inline packages/integration-tests/src/tests/ui/applications/helpers.ts:64]: Feels like a reusable util method, sth like expectToHaveModelWithTitle

File: packages/integration-tests/src/tests/ui/applications/application-test-cases.ts (hunk 1 above)
Comment (charIeszhao) [reply packages/integration-tests/src/tests/ui/applications/application-test-cases.ts:18]: Maybe we can consider copy the metadata ts file from console?

File: packages/integration-tests/src/tests/ui/applications/helpers.ts (hunk 2 above)
Comment (xiaoyijun) [reply packages/integration-tests/src/tests/ui/applications/helpers.ts]: Updated.

File: packages/integration-tests/src/tests/ui/applications/helpers.ts (hunk 3 above)
Comment (xiaoyijun) [reply packages/integration-tests/src/tests/ui/applications/helpers.ts:64]: Will update in a later PR.

Comment (xiaoyijun) [reply packages/integration-tests/src/tests/ui/applications/helpers.ts]: Updated.

File: packages/integration-tests/src/tests/ui/applications/helpers.ts (hunk 2 above)
Comment (xiaoyijun) [reply packages/integration-tests/src/tests/ui/applications/helpers.ts]: Updated.

File: packages/integration-tests/src/tests/ui/applications/typed-application-test-cases.ts [hunk 4]
```diff
@@ -0,0 +1,33 @@
...
+  description: 'This is a traditional web app',
+};
+
+const machineToMachine: TypedApplicationCase = {
+  type: ApplicationType.MachineToMachine,
+  name: 'Machine to Machine App',
+  description: 'This is a machine to machine app',
+};
+
+export const typedApplicationCases = [native, spa, traditional, machineToMachine] as const;
```
Comment (charIeszhao) [inline packages/integration-tests/src/tests/ui/applications/typed-application-test-cases.ts:33]: We can iterate `ApplicationType enum` and dynamically generate a metadata object

File: packages/integration-tests/src/tests/ui/applications/index.test.ts [hunk 5]
```diff
@@ -0,0 +1,269 @@
...
+    });
+
+    await expectModalWithTitle(page, 'Start with SDK and guides');
+
+    await expect(page).toClick(
+      '.ReactModalPortal div[class$=header] button[class$=requestSdkButton] span',
+      {
+        text: "Can't find your guide?",
+      }
+    );
```
Comment (charIeszhao) [inline packages/integration-tests/src/tests/ui/applications/index.test.ts:100]: Can we find the button without matching button text?

**Review by charIeszhao — APPROVED** [review]

File: packages/integration-tests/src/tests/ui/applications/typed-application-test-cases.ts (hunk 4 above)
Comment (xiaoyijun) [reply packages/integration-tests/src/tests/ui/applications/typed-application-test-cases.ts]: updated.

File: packages/integration-tests/src/tests/ui/applications/index.test.ts (hunk 5 above)
Comment (xiaoyijun) [reply packages/integration-tests/src/tests/ui/applications/index.test.ts]: Updated.

File: packages/integration-tests/src/tests/ui/applications/application-test-cases.ts (hunk 1 above)
Comment (xiaoyijun) [reply packages/integration-tests/src/tests/ui/applications/application-test-cases.ts]: Updated.
Per offline discussion, these use cases are generated by templates,so we only need to select one for testing.

**Pull request merged**